integration_test/tests/prepared_statement_tests.rs

```diff
@@ -0,0 +1,186 @@
+mod common;
+
+mod integration {
+    use crate::common::{
+        allocate_env, allocate_statement, connect_with_conn_string, disconnect_and_close_handles,
+        fetch_and_get_data, get_column_attributes, get_sql_diagnostics,
+    };
+    use odbc_sys::{
+        CDataType, HDbc, HStmt, Handle, HandleType, SQLExecute, SQLFetch, SQLPrepareW, SmallInt,
+        SqlReturn, NTS,
+    };
+
+    use cstr::WideChar;
+
+    fn connect_and_allocate_statement() -> (HDbc, HStmt) {
+        let env_handle = allocate_env().unwrap();
+        let conn_str = crate::common::generate_default_connection_str();
+        let conn_handle = connect_with_conn_string(env_handle, conn_str).unwrap();
+        (conn_handle, allocate_statement(conn_handle).unwrap())
+    }
+
+    #[test]
+    fn test_error_execute_before_prepare() {
+        let (dbc, stmt) = connect_and_allocate_statement();
+
+        let mut query: Vec<WideChar> = cstr::to_widechar_vec("select * from example");
+        query.push(0);
+        unsafe {
+            // Only prepared statement can be executed.
+            // Calling SQLExecute before SQLPrepare is invalid.
+            assert_eq!(SqlReturn::ERROR, SQLExecute(stmt));
+
+            dbg!(get_sql_diagnostics(HandleType::Stmt, stmt as Handle));
```

Comment:
should we remove the dbg! from this and the one below?

Reply:
I added the dbg on purpose because we could get a SQL_ERROR for a different reason than the one we think is happening. Let me see if we have a helper to check error message/code themselves.